### PR TITLE
Contributing Enhancements and Guidelines

### PR DESCRIPTION
**Pull request by adampickeral**
Added contributing guidelines.

Created scripts for running various development tasks that follow github's script guidelines for easier setup and development:
scripts/cibuild - runs lint, build, and test for CI server
scripts/server - starts an http server to serve up demo components; also watches for file changes
scripts/setup - installs all dependencies needed for development
scripts/test - starts karma to run unit tests; watches for file changes
scripts/update - can be run after pulling in master to get latest dependencies

### CODE REVIEW

**Comment by willcumbie** [comment]
Contributor guidelines look solid. Scripts aren't working for me at the moment, probably as a result of a poor wifi connection; I'll try them again this evening and update.

File: README.md [hunk 1]
```diff
@@ -6,16 +6,13 @@ CAUTION: This library is not ready for public use yet.
...
-### Building
-To build a bundled file for release run `npm run build`
-To build a demo bundle for testing run `npm run demo-build` and open `canon-react/demo/index.html` in your browser
+## Building
+The latest stable release is available under the releases section of the repository. You can also generate a release from master if you wish to live on the edge.
 
-### Testing
-To run tests with phantom run `npm run test`
-To run tests in chrome and watch for changes run `npm run test-watch`
+To build a bundled file for release run `scripts/cibuild`
```
Comment (underbyte) [inline README.md]: shouldn't this script be `scripts/test`? we have an alias for `npm run test`, I think that should be here too

File: README.md [hunk 2]
```diff
@@ -6,16 +6,13 @@ CAUTION: This library is not ready for public use yet.
...
-## Instructions For Devs
-Git clone and run `npm install` to install dependencies (may need to be run as sudo)
+## Contributing
+For issues or feature requests please submit an [issue](https://github.com/rackerlabs/canon-react/issues). For more information on contributing to canon-react, checkout the [Contributing Guide](CONTRIBUTING.md)
 
-### Building
-To build a bundled file for release run `npm run build`
-To build a demo bundle for testing run `npm run demo-build` and open `canon-react/demo/index.html` in your browser
+## Building
+The latest stable release is available under the releases section of the repository. You can also generate a release from master if you wish to live on the edge.
```
Comment (underbyte) [inline README.md]: don't we wanna keep `npm run build` here?

**Comment by underbyte** [comment]
For the most part, everything looks good!

File: README.md (hunk 1 above)
Comment (adampickeral) [inline README.md]: not sure what this comment is referencing, testing info is in the contributing file

File: README.md (hunk 2 above)
Comment (adampickeral) [inline README.md]: no. the idea of using these scripts is that they hide the details of any tooling used to do setup, testing, build, etc.

http://githubengineering.com/scripts-to-rule-them-all/

File: README.md (hunk 1 above)
Comment (underbyte) [inline README.md]: ignore this thought it was referring to testing section

**Comment by willcumbie** [comment]
Scripts run fine. LGTM.

**Pull request merged**